gui/src/app/StanSampler/StanSampler.ts

```diff
@@ -0,0 +1,85 @@
+import { Progress, Replies, Requests } from '../tinystan/Worker';
+import StanWorker from '../tinystan/Worker?worker';
+
+export type StanSamplerStatus = '' | 'loading' | 'loaded' | 'sampling' | 'completed' | 'failed';
+
+class StanSampler {
+    #worker: Worker | undefined;
+    #status: StanSamplerStatus = '';
+    #errorMessage: string = '';
+    #onProgressCallbacks: ((progress: Progress) => void)[] = [];
+    #onStatusChangedCallbacks: (() => void)[] = [];
+    #draws: number[][] = [];
+    #paramNames: string[] = [];
+    constructor(private compiledUrl: string) {
+        this._initialize()
+    }
+    _initialize() {
+        this.#worker = new StanWorker
+        this.#status = 'loading'
+        this.#worker.onmessage = (e) => {
+            const purpose: Replies = e.data.purpose;
+            switch (purpose) {
+                case Replies.Progress: {
+                    this.#onProgressCallbacks.forEach(callback => callback(e.data.report));
+                    break;
+                }
+                case Replies.ModelLoaded: {
+                    this.#status = 'loaded';
+                    this.#onStatusChangedCallbacks.forEach(cb => cb())
+                    break;
+                }
+                case Replies.SampleReturn: {
+                    if (e.data.error) {
+                        this.#errorMessage = e.data.error;
+                        this.#status = 'failed';
+                        this.#onStatusChangedCallbacks.forEach(cb => cb())
+                    } else {
+                        this.#draws = e.data.draws;
+                        this.#paramNames = e.data.paramNames;
+                        this.#status = 'completed';
+                        this.#onStatusChangedCallbacks.forEach(cb => cb())
+                    }
+                    break;
+                }
+            }
+        }
+        this.#worker.postMessage({ purpose: Requests.Load, url: this.compiledUrl });
+    }
+    sample(a: {data: any, numChains: number}) {
```

Comment:
The right type to accept here would be `SamplerParams` from TinyStan/index.ts and just pass it unchanged, I think

Reply:
I have made this change

Reply:
Ooops, the name of my commit message was "i have made this change" when that was supposed to be my comment in this issue. Please forgive. :)

Reply:
I think I misled you, it should be `Partial<SamplerParams>`, which lets `defaultSamplerParams` stay private (which it would be if/when `tinystan.ts` ended up in its own package